src/pages/workspaces/workspace/Analyses.js

```diff
@@ -260,14 +237,15 @@ const Analyses = _.flow(
     Utils.withBusyState(setBusy)
   )(async () => {
     const notebooks = await Ajax(signal).Buckets.listNotebooks(namespace, bucketName)
-    const enhancedNotebooks = _.map(notebook => _.merge(notebook, { toolLabel: tools.Jupyter.label }), notebooks)
+    //we map the `toolLabel` and `updated` fields to their corresponding header label, which simplifies the table sorting code
+    const enhancedNotebooks = _.map(notebook => _.merge(notebook, { application: tools.Jupyter.label, lastModified: notebook.updated }), notebooks)
```

Comment:
nit: Could you make the ajax calls concurrent?

```javascript
    const [notebooks, rmds] = await Promise.all([
      Ajax(signal).Buckets.listNotebooks(namespace, bucketName),
      Ajax(signal).Buckets.listRmds(namespace, bucketName)
    ])
    //we map the `toolLabel` and `updated` fields to their corresponding header label, which simplifies the table sorting code
    const enhancedNotebooks = _.map(notebook => _.merge(notebook, { application: tools.Jupyter.label, lastModified: notebook.updated }), notebooks)
    const enhancedRmd = _.map(rmd => _.merge(rmd, { application: tools.RStudio.label, lastModified: rmd.updated }), rmds)
```

Looking at the ajax calls - it looks like we are making the same call exact call, and filtering at the Ajax call level.
Could you add a ticket for updating this approach so we can explore a some different abstractions?

Reply:
I can probably just change where the filtering happens here and use one Ajax call tbh

Reply:
Thanks, that looks better, and one call is cleaner